tests/read_mac/test_read_mac_metadata.py

```diff
@@ -0,0 +1,146 @@
+        self.image1 = request.config.getoption("--image1")
+        self.image2 = request.config.getoption("--image2")
+        self.minigraph1 = request.config.getoption("--minigraph1")
+        self.minigraph2 = request.config.getoption("--minigraph2")
+
+        if not self.image1:
+            pytest.fail("Please specify --image1")
+        if not self.image2:
+            pytest.fail("Please specify --image2")
+        if not self.iteration or self.iteration < 1: 
+            pytest.fail("Please specify --iteration in correct range")
+
+        self.request = request
+        duthost = request.getfixturevalue('duthost')
+
+        minigraph_exist = duthost.stat(path = "/etc/sonic/minigraph.xml")["stat"]["exists"]
+        if (self.minigraph1 is not None or self.minigraph2 is not None) and minigraph_exist:
+            logger.info("Store current minigraph for debug purpose")
+            duthost.shell("cp /etc/sonic/minigraph.xml /etc/sonic/minigraph.xml.backup")
+
+    def run_test_in_reinstall_loop(self):
+        logger.info("Verify MAC in image reinstall loop")
+        duthost  = self.request.getfixturevalue('duthost')
+        localhost = self.request.getfixturevalue('localhost')
+
+        for counter in range(1, self.iteration + 1):
+            current_minigraph = None
+
+            if counter % 2 == 1:
+                current_image = self.image1
+                current_minigraph = self.minigraph1
+            else:
+                current_image = self.image2
+                current_minigraph = self.minigraph2
+
+            logger.info("Iteration #{}".format(counter))
+            logger.info("Current image: {}, temporary minigraph: {}".format(current_image, current_minigraph))
+            if current_minigraph:
+                logger.info("Copy specified minigraph {} to the /etc/sonic folder".format(current_minigraph))
+                duthost.copy(src=current_minigraph, dest="/etc/sonic/minigraph.xml")
+
+            loganalyzer = LogAnalyzer(ansible_host=duthost, marker_prefix="read_mac_metadata")
+            loganalyzer.match_regex = [ ".*can't parse mac address 'None'*" ]
+
+            with loganalyzer:
+                self.deploy_image_to_duthost(duthost, localhost , current_image)
+                self.reboot_dut(localhost, duthost)
+
+            if current_minigraph:
+                logger.info("Execute cli 'config load_minigraph -y' to apply new minigraph")
+                duthost.shell("config load_minigraph -y")
+                duthost.shell("config save -y")
+
+            logger.info("Remove old (not current) sonic image")
+            duthost.reduce_and_add_sonic_images(disk_used_pcent = 1)
+
+            self.check_mtu_and_interfaces(duthost)
+
+    def deploy_image_to_duthost(self, duthost, localhost, current_image):
+        logger.info("Download SONiC image {}".format(current_image))
+        localhost.get_url(url=current_image, dest=BINARY_FILE_ON_LOCALHOST)
```

Comment:
Suggest to move the image downloading part to test case setup part. Since `deploy_image_to_duthost` is called inside the iteration, the `get_url` will be called multiple times. It's unnecessary and time consuming.

Reply:
Moved to __init__ . Now images are downloading  only once.

About moving to "setup" - I decided move to "__init__" to prevent huge design changes in this test